django_lti_tool_provider/views.py

```diff
@@ -67,41 +77,33 @@ def process_request(self, request):
             return self.process_anonymous_lti(request)
 
     @classmethod
-    def _get_lti_parameters_from_request(cls, request):
-        provider = DjangoToolProvider(settings.LTI_CLIENT_KEY, settings.LTI_CLIENT_SECRET, request.POST)
-        provider.valid_request(request)
-        return provider.to_params()
-
-    @classmethod
-    def _store_lti_parameters(cls, user, parameters):
-        """
-        Filters out OAuth parameters than stores LTI parameters into the DB, creating or updating record as needed
-        """
-        # TODO: this method got complicated - move into model/model manager
-        lti_params = {
+    def lti_param_filter(cls, parameters):
+        return {
             key: value
             for key, value in parameters.iteritems()
             if 'oauth' not in key
         }
-        custom_key = cls.authentication_manager.vary_by_key(lti_params)
-
-        # implicitly tested by test_views
-        if custom_key is None:
-            custom_key = ''
-
-        lti_user_data, created = LtiUserData.objects.get_or_create(user=user, custom_key=custom_key)
-        if not created:
-            _logger.debug(u"Replaced LTI parameters for user %s", user.username)
-
-        if lti_user_data.edx_lti_parameters.get('user_id', lti_params['user_id']) != lti_params['user_id']:
-            # TODO: not covered by test
-            message = u"LTI parameters for user found, but anonymous user id does not match. " \
-                      u"This might be caused by stale cookies. Clear browser cookies and try again"
-            _logger.error(message)
-            raise ValueError(message)
-        lti_user_data.edx_lti_parameters = lti_params
-        lti_user_data.save()
-        return lti_user_data
+
+    @classmethod
+    def _right_user(cls, user, lti_parameters):
+        try:
+            info, created = LtiUserData.get_or_create_by_parameters(
+                user, cls.authentication_manager, cls.lti_param_filter(lti_parameters)
+            )
+            if created:
+                # If this is the first time the user's data is being created, that means
+                # that the user predated the LTI request.
+                info.delete()
```

Comment:
@Kelketek In this branch three DB requests are performed, where one would suffice. Might make sense to split that `get_or_create_by_parameters` into `get_by_parameters` and `create`, maybe?